### PR TITLE
DOC: integrate: Add complex matrix DE solution to examples

### LINKED ISSUE #13437: DOC: Add example as to how to use solve_ivp to solve complex matrix differential equations
**Is your feature request related to a problem? Please describe.**
No. 
The example given in the documentation does not make it easy to 
guess the way to use solve_ivp to solve complex matrix differential equations.

**Describe the solution you'd like**
Add a sufficiently  detailed example

**Describe alternatives you've considered**
[example ](https://www.dropbox.com/s/ip1fqko2f7xfs41/patol75_answer_pristine_only_matrix.ipynb?dl=0) based on [this answer](https://stackoverflow.com/a/65838993/7952027) to this [question](https://stackoverflow.com/q/65793457/7952027)

**Additional context (e.g. screenshots)**

### PR DESCRIPTION
**Pull request by topolskikevin**
#### Reference issue

Closes #13437

#### What does this implement/fix?

Added examples to solve complex matrix differential equations to docstring

#### Additional information

First commit, welcome to any feedback and changes I can follow up.

### CODE REVIEW

File: scipy/integrate/_ivp/ivp.py [hunk 1]
```diff
@@ -426,6 +426,46 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
      [8.         7.5534414  4.25308709 1.73277247 0.72069014 0.29932181
       0.12428631 0.05403123]]
 
+    A couple examples of using solve_ivp to solve differential functions with
+    complex matrix A
+
+    >>> import numpy as np
```
Comment (tupuinui) [inline scipy/integrate/_ivp/ivp.py:432]: We don't import NumPy as this is done for all examples behind the scene.
```suggestion
```

File: scipy/integrate/_ivp/ivp.py [hunk 2]
```diff
@@ -426,6 +426,46 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
      [8.         7.5534414  4.25308709 1.73277247 0.72069014 0.29932181
       0.12428631 0.05403123]]
 
+    A couple examples of using solve_ivp to solve differential functions with
+    complex matrix A
```
Comment (tupuinui) [inline scipy/integrate/_ivp/ivp.py:430]:
```suggestion
    complex matrix ``A``.
```

File: scipy/integrate/_ivp/ivp.py [hunk 3]
```diff
@@ -426,6 +426,46 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
+
+    >>> import numpy as np
+    >>> from scipy.integrate import solve_ivp
+    >>> A = np.array([[-0.25 + 0.14j, 0, 0.33 + 0.44j],
+    ...               [0.25 + 0.58j, -0.2 + 0.14j, 0],
+    ...               [0, 0.2 + 0.4j, -0.1 + 0.97j]])
+
+    Solving an ivp with A from above and y as 3x1 vector:
```
Comment (tupuinui) [inline scipy/integrate/_ivp/ivp.py:438]:
```suggestion
    Solving an ivp with ``A`` from above and ``y`` as 3x1 vector:
```

File: scipy/integrate/_ivp/ivp.py [hunk 4]
```diff
@@ -426,6 +426,46 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
+    ...     return A @ y
+    >>> result = solve_ivp(deriv_vec, [0, 25],
+    ...                    np.array([10 + 0j, 20 + 0j, 30 + 0j]),
+    ...                    t_eval=np.linspace(0, 25, 101)
+    >>> print(result.y[:, 0])
+    [10.+0.j 20.+0.j 30.+0.j]
+    >>> print(result.y[:, -1])
+    [18.46+45.25j 10.01+36.23j -4.98+80.07j]
+
+    Solving an ivp with A from above y as 3x3 matrix :
```
Comment (tupuinui) [inline scipy/integrate/_ivp/ivp.py:450]:
```suggestion
    Solving an ivp with ``A`` from above ``y`` as 3x3 matrix :
```

**Review by tupuinui — CHANGES_REQUESTED** [review]
Thank you for your first PR @topolskikevin! Nice example.

On my side (as I don't know these functions) I just have a few cosmetic comments. Let see as well if the doctests pass.

File: scipy/integrate/_ivp/ivp.py (hunk 2 above)
Comment (tupui) [inline scipy/integrate/_ivp/ivp.py:430]:
```suggestion
    complex matrix ``A``.
```

File: scipy/integrate/_ivp/ivp.py (hunk 1 above)
Comment (tupui) [inline scipy/integrate/_ivp/ivp.py:432]: We don't import NumPy as this is done for all examples behind the scene.
```suggestion
```

File: scipy/integrate/_ivp/ivp.py (hunk 3 above)
Comment (tupui) [inline scipy/integrate/_ivp/ivp.py:438]:
```suggestion
    Solving an ivp with ``A`` from above and ``y`` as 3x1 vector:
```

File: scipy/integrate/_ivp/ivp.py (hunk 4 above)
Comment (tupui) [inline scipy/integrate/_ivp/ivp.py:450]:
```suggestion
    Solving an ivp with ``A`` from above ``y`` as 3x3 matrix :
```

**Review by tupui — CHANGES_REQUESTED** [review]
Sorry about the mess, I used the wrong account... (Believe me, I hate to have this setup but I don't have the choice...)

File: scipy/integrate/_ivp/ivp.py (hunk 2 above)
Comment (topolskikevin) [reply scipy/integrate/_ivp/ivp.py]: Thanks Pamphile! And no worries with the set up! I have made the changes!

File: scipy/integrate/_ivp/ivp.py [hunk 5]
```diff
@@ -426,6 +426,45 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
+    ...               [0.25 + 0.58j, -0.2 + 0.14j, 0],
+    ...               [0, 0.2 + 0.4j, -0.1 + 0.97j]])
+
+    Solving an ivp with ``A`` from above and ``y`` as 3x1 vector:
+
+    >>> def deriv_vec(t, y):
+    ...     return A @ y
+    >>> result = solve_ivp(deriv_vec, [0, 25],
+    ...                    np.array([10 + 0j, 20 + 0j, 30 + 0j]),
+    ...                    t_eval=np.linspace(0, 25, 101)
```
Comment (tupui) [inline scipy/integrate/_ivp/ivp.py:443]:
```suggestion
    ...                    t_eval=np.linspace(0, 25, 101))
```

File: scipy/integrate/_ivp/ivp.py [hunk 6]
```diff
@@ -426,6 +426,45 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
+    >>> result = solve_ivp(deriv_mat, [0, 25], y0.flatten(),
+    ...                    t_eval=np.linspace(0, 25, 101))
+    >>> print(result.y[:, 0].reshape(3, 3))
+    [[ 2.+0.j  3.+0.j  4.+0.j]
+     [ 5.+0.j  6.+0.j  7.+0.j]
+     [ 9.+0.j 34.+0.j 78.+0.j]]
+    >>> print(result.y[:, -1].reshape(3, 3))
+    [[ 5.67+12.07j 17.28+31.03j 37.83+63.25j]
+     [ 3.39+11.82j 21.32+44.88j 53.17+103.80j]
+     [ -2.26+22.19j -15.12+70.191j -38.34+153.29j]]
```
Comment (tupui) [inline scipy/integrate/_ivp/ivp.py:466]: Doctests are very picky. It must be formatted exactly the same way. You could use `...` in numbers to round if you want.
```suggestion
    [[  5.67451179 +12.07938445j  17.2888073  +31.03278837j
       37.83405768 +63.25138759j]
     [  3.39949503 +11.82123994j  21.32530996 +44.88668871j
       53.17531184+103.80400411j]
     [ -2.26105874 +22.19277664j -15.1255713  +70.19616341j
      -38.34616845+153.29039931j]]
```

**Review by tupui — CHANGES_REQUESTED** [review]
Thanks for the update. You still have issues with the CI. These suggestions should fix them.

File: scipy/integrate/_ivp/ivp.py [hunk 7]
```diff
@@ -426,6 +426,49 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
+     [ 9.+0.j 34.+0.j 78.+0.j]]
+    >>> print(result.y[:, -1].reshape(3, 3))
+    [[  5.67451179 +12.07938445j  17.2888073  +31.03278837j
+        37.83405768 +63.25138759j]
+     [  3.39949503 +11.82123994j  21.32530996 +44.88668871j
+        53.17531184+103.80400411j]
+     [ -2.26105874 +22.19277664j -15.1255713  +70.19616341j
+       -38.34616845+153.29039931j]]
+
     Specifying points where the solution is desired.
```
Comment (tupui) [inline scipy/integrate/_ivp/ivp.py:472]: The following example should be placed above as the `exponential_decay` function is defined above. The examples are linked.

File: scipy/integrate/_ivp/ivp.py [hunk 8]
```diff
@@ -426,6 +426,49 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
      [8.         7.5534414  4.25308709 1.73277247 0.72069014 0.29932181
       0.12428631 0.05403123]]
 
+    A couple examples of using solve_ivp to solve differential functions with
+    complex matrix ``A``.
+
+    >>> from scipy.integrate import solve_ivp
```
Comment (tupui) [inline scipy/integrate/_ivp/ivp.py:432]: Already imported above.
```suggestion
```

**Review by tupui — CHANGES_REQUESTED** [review]

**Comment by tupui** [comment]
@nmayorov could you have a look at the example itself?

**Comment by Patol75** [comment]
@tupui The idea of this PR is to provide an example of how to use `solve_ivp` when the variable `y` is a matrix-valued function. In other words, how to solve an ordinary differential equation that involves a matrix-valued function through `solve_ivp`. The motivation behind this PR is this [SO question](https://stackoverflow.com/q/65793457/10640534) that I have answered in which the author is confused about how to proceed in the aforementioned scenario. Indeed, throughout the documentation, it is mentioned that `y` is a vector. However, the confusion comes from the fact that the author assumed it was necessary to pass a 2-D NumPy array to `solve_ivp`, while the way to proceed is to provide `solve_ivp` with the flattened array and reshape `y` within `fun` to perform matrix operations before flattening again the returned array. Following my answer on SO, the question's author opened #13437 to ask for a similar example to be included in the documentation. The present PR simply reproduces my answer to the author's question on SO. The ordinary differential equation used does not have anything special. Instead, the idea is to show the flatten/reshape procedure to allow for execution through `solve_ivp`. I guess that the question here is whether SciPy developers consider this scenario should be documented, and if yes, whether the proposed approach is satisfying.

**Comment by mdhaber** [comment]
@nmayorov Do you have a moment to check out this new example?

**Comment by nitin-pandita** [comment]
Can i work on it

**Comment by tupui** [comment]
Hi @nitin-pandita, I am afraid there is no simple way to work on someone else's PR. I would suggest you look at other issues.

**Comment by nitin-pandita** [comment]
> Hi @nitin-pandita, I am afraid there is no simple way to work on someone else's PR. I would suggest you look at other issues.

Ok Sure

File: scipy/integrate/_ivp/ivp.py [hunk 9]
```diff
@@ -503,6 +503,49 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
     >>> plt.title('Lotka-Volterra System')
     >>> plt.show()
 
+    A couple examples of using solve_ivp to solve differential functions with
+    complex matrix ``A``.
```
Comment (j-bowhay) [inline scipy/integrate/_ivp/ivp.py:507]:
```suggestion
    A couple examples of using solve_ivp to solve the differential equation ``y'=Ay``
    with complex matrix ``A``.
```

File: scipy/integrate/_ivp/ivp.py [hunk 10]
```diff
@@ -503,6 +503,49 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
+
+    >>> A = np.array([[-0.25 + 0.14j, 0, 0.33 + 0.44j],
+    ...               [0.25 + 0.58j, -0.2 + 0.14j, 0],
+    ...               [0, 0.2 + 0.4j, -0.1 + 0.97j]])
+
+    Solving an ivp with ``A`` from above and ``y`` as 3x1 vector:
```
Comment (j-bowhay) [inline scipy/integrate/_ivp/ivp.py:513]:
```suggestion
    Solving an IVP with ``A`` from above and ``y`` as 3x1 vector:
```

File: scipy/integrate/_ivp/ivp.py [hunk 11]
```diff
@@ -503,6 +503,49 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
+    >>> result = solve_ivp(deriv_vec, [0, 25],
+    ...                    np.array([10 + 0j, 20 + 0j, 30 + 0j]),
+    ...                    t_eval=np.linspace(0, 25, 101))
+    >>> print(result.y[:, 0])
+    [10.+0.j 20.+0.j 30.+0.j]
+    >>> print(result.y[:, -1])
+    [18.46291039+45.25653651j 10.01569306+36.23293216j
+     -4.98662741+80.07360388j]
+
+    Solving an ivp with ``A`` from above ``y`` as 3x3 matrix :
```
Comment (j-bowhay) [inline scipy/integrate/_ivp/ivp.py:526]:
```suggestion
    Solving an IVP with ``A`` from above ``y`` as 3x3 matrix :
```

File: scipy/integrate/_ivp/ivp.py [hunk 12]
```diff
@@ -520,6 +520,49 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
...
+    >>> result = solve_ivp(deriv_vec, [0, 25],
+    ...                    np.array([10 + 0j, 20 + 0j, 30 + 0j]),
+    ...                    t_eval=np.linspace(0, 25, 101))
+    >>> print(result.y[:, 0])
+    [10.+0.j 20.+0.j 30.+0.j]
+    >>> print(result.y[:, -1])
+    [18.46291039+45.25653651j 10.01569306+36.23293216j
+     -4.98662741+80.07360388j]
+
+    Solving an IVP with ``A`` from above ``y`` as 3x3 matrix :
```
Comment (j-bowhay) [inline scipy/integrate/_ivp/ivp.py:543]:
```suggestion
    Solving an IVP with ``A`` from above with ``y`` as 3x3 matrix :
```

File: scipy/integrate/_ivp/ivp.py [hunk 13]
```diff
@@ -520,6 +520,49 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
     >>> plt.title('Lotka-Volterra System')
     >>> plt.show()
 
+    A couple examples of using solve_ivp to solve the differential equation ``y'=Ay``
```
Comment (j-bowhay) [inline scipy/integrate/_ivp/ivp.py:523]:
```suggestion
    A couple examples of using solve_ivp to solve the differential equation ``y' = Ay``
```

File: scipy/integrate/_ivp/ivp.py [hunk 14]
```diff
@@ -520,6 +520,49 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
     >>> plt.title('Lotka-Volterra System')
     >>> plt.show()
 
+    A couple examples of using solve_ivp to solve the differential equation ``y' = Ay``
+    with complex matrix ``A``.
```
Comment (j-bowhay) [inline scipy/integrate/_ivp/ivp.py:524]:
```suggestion
    A couple examples of using solve_ivp to solve the differential
    equation ``y' = Ay`` with complex matrix ``A``.
```

**Review by j-bowhay — APPROVED** [review]
Thanks @topolskikevin

**Pull request merged**